src/redux/books/books.js

```diff
@@ -13,28 +16,66 @@ export const remove = (index) => ({
   payload: index,
 });
 
-const initialState = [{
-  id: 1,
-  title: 'book1',
-  author: 'author1',
-},
-{
-  id: 2,
-  title: 'book2',
-  author: 'author2',
-},
-];
+export const read = (books) => ({
+  type: READ,
+  payload: books,
+});
 
 // Reducer
-const booksReducer = (state = initialState, action) => {
+const booksReducer = (state = [], action) => {
   switch (action.type) {
+    case READ:
+      return action.payload;
     case ADD:
       return [...state, action.payload];
     case REMOVE:
-      return [...state.filter((item) => item.id !== action.payload)];
+      return [...state.filter((item) => item.item_id !== action.payload)];
     default:
       return state;
   }
 };
 
+/* eslint-disable array-callback-return */
+export const recieveBooks = () => async (dispatch) => {
+  await fetch(APIURL)
+    .then((res) => res.json())
+    .then((books) => {
+      const bookList = [];
+      Object.keys(books).map((key) => {
+        bookList.push({
+          item_id: key,
+          title: books[key][0].title,
+          author: books[key][0].author,
+          category: books[key][0].category,
+        });
+      });
+      dispatch(read(bookList));
+    });
+};
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.), For getting book data you suppose to use `createAsyncThunk`, So please fix that.
see this [article ](https://blog.logrocket.com/using-redux-toolkits-createasyncthunk/)to understand more about it.
[Hint] You can import `createAsyncThunk` from `@reduxjs/toolkit `library and use it like that:
```javascript
export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```